internal/librarian/update.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package librarian
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/googleapis/librarian/internal/config"
+	"github.com/googleapis/librarian/internal/fetch"
+	"github.com/googleapis/librarian/internal/yaml"
+	"github.com/urfave/cli/v3"
+)
+
+var (
+	githubAPI      = "https://api.github.com"
+	githubDownload = "https://github.com"
+	sourceRepos    = map[string]*fetch.Repo{
+		"googleapis": {Org: "googleapis", Repo: "googleapis"},
+		"discovery":  {Org: "googleapis", Repo: "discovery-artifact-manager"},
+	}
+)
+
+// updateCommand returns the `update` subcommand.
+func updateCommand() *cli.Command {
+	return &cli.Command{
+		Name:      "update",
+		Usage:     "update sources to the latest version",
+		UsageText: "librarian update [--all | source]",
+		Flags: []cli.Flag{
+			&cli.BoolFlag{
+				Name:  "all",
+				Usage: "update all sources",
+			},
+		},
+		Action: func(ctx context.Context, cmd *cli.Command) error {
+			all := cmd.Bool("all")
+			source := cmd.Args().First()
+
+			if all && source != "" {
+				return fmt.Errorf("cannot specify a source when --all is set")
+			}
+			if !all && source == "" {
+				return fmt.Errorf("a source must be specified, or use the --all flag")
+			}
+			return runUpdate(all, source)
+		},
+	}
+}
+
+func runUpdate(all bool, sourceName string) error {
+	cfg, err := yaml.Read[config.Config](librarianConfigPath)
+	if err != nil {
+		return err
+	}
+	if cfg.Sources == nil {
+		return errEmptySources
+	}
+
+	endpoints := &fetch.Endpoints{
+		API:      githubAPI,
+		Download: githubDownload,
+	}
+
+	sourcesMap := map[string]*config.Source{
+		"googleapis": cfg.Sources.Googleapis,
+		"discovery":  cfg.Sources.Discovery,
+	}
+
```

Comment:
super nit (feel free to ignore, just sharing how I think): I typically only add whitespace when there is something to convey, so that the whitespace is meaningful. In that case, I wouldn't put whitespace here or on line 103, because it just makes the file longer without conveying anything meaningful